### ISSUE TITLE
Make master a protected branch

### ISSUE THREAD

**Issue by tawashley** [code-computerlove/node-static-site-generator#52]
GitHub allows you to make specific branches protected. Wonder if I could open this for discussion,

Allow will disabling direct pushes to a branch, it enables you to check status checks pass before being able to merge. 

This would allow us to enforce the workflow of: 
* Branch off from master
* Do the codez
* Submit Pull Request for master
* Run status checks - check it installs correctly, tests are green
* Merge new code into master

This looks like it could help, especially in areas such as Windows support etc. 

@andrewbrandwood @bmds @furzeface @code-mattclaffey

**Comment by furzeface** [comment]
Yeah a few of us on an open source project were discussing doing this yesterday for our project, I'm not sure how it's done though. It would be useful I think.

**Comment by tawashley** [comment]
To enable this feature:

* (Repo) Settings - right most tab
* Branches
* Protected Branches

**Comment by furzeface** [comment]
Yeah I've just tested it out and it works perfectly! A **BIG** :+1: to this @tawashley :)

**Comment by tawashley** [comment]
I'm going to go ahead and enable this. :bomb:

**Comment by tawashley** [comment]
Enabled. I'm going to leave this open in the short-term.
